docs/adrs/0001-support-caching-deps-for-monorepos.md

```diff
@@ -0,0 +1,48 @@
+# 0. Support caching dependencies for mono repos
+Date: 2021-07-13
+
+Status: Proposed
+
+## Context
+Currently, `actions/setup-node` supports caching dependencies for Npm and Yarn package managers.  
+For the first iteration, we have decided to not support cases where `package-lock.json` / `yarn.lock` are located outside of repository root.  
+Current implementation searches the following file patterns in the repository root: `package-lock.json`, `yarn.lock` (in order of resolving priorities)
+
+Obviously, it made build-in caching unusable for mono-repos and repos with complex structure.  
+We would like to revisit this decision and add customization for dependencies lock file location.
+
+## Proposal
+We have the following options:
+1. Allow to specify  directory where `package-lock.json` or `yarn.lock` are located
+2. Allow to specify path to the dependencies lock file (including directory path and filename)
```

Comment:
Will option 2 be modeled similar to the 'path' keyword used in actions/cache?

Reply:
Nope, `path` input in actions/cache defines what folder will be cached. Users of setup-node don't need to care about it. It is done under hood from our side.
This is more about postfix of `key` input, see: 
> key: ${{ runner.os }}-${{ hashFiles('**src/package-lock.json**') }}

User specifies path to file that describes their dependencies and hash of this file will be used as a part of key.

Reply:
I meant, will the input syntax and structure of option 2 be similar to what we currently provide with 'path'. 

```
cache-dependency-path: |
  - xyz/requirements.txt
  - abc/requirements.txt
```

Reply:
Hm, I think wildcard will resolve such cases too but I don't have objections to support both options:
```yml
cache-dependency-path: xyz/requirements.txt
```
or
```yml
cache-dependency-path: |
  xyz/requirements.txt
  abc/requirements.txt
```